source/MissionAction.h

```diff
@@ -41,6 +43,18 @@ class UI;
 // In addition to performing a GameAction, a MissionAction can gate the task on
 // the ownership of specific outfits and also display dialogs or conversations.
 class MissionAction {
+private:
+	struct MissionDialog {
```

Comment:
```suggestion
	class MissionDialog {
	public:
```

Reply:
This is a simple bucket of data; a tuple. It isn't intended to be treated as a class with private data. It is a storage mechanism for the parent. Hence, it is a struct.

Reply:
> Use of class vs. struct
> ▽ Do not use the "struct" keyword; just define a class with public data members instead.
> Exception: occasionally when using the POSIX libraries you need the "struct" keyword to define that a typename is a struct and not a function of the same name, e.g.:
> ```c++
> struct stat sb;
> stat("path/to/file", &sb);
> ```

~~I do not really care about your arguments, it is in the style guide~~

It does make no difference to the compiler to do one or the other (at least no meaningful difference in our case) so please just change it for coherence

Reply:
If I make it a class, we head in the needless direction of making the members private and giving them accessor and mutator functions. Then the code changes will be fifty lines longer and less readable. A misguided academic elegance that serves nobody.

Reply:
if you want it to be a tuple, use a tuple

Reply:
I've made the change. It is now a class with private members, and its containing class is a friend.

Reply:
Or, just make the fields public as the part of the style guide quoted by RisingLeaf mentioned:

> just define a class with public data members instead.

Reply:
Now I am sorry, but this change is not what I asked for.
I just wanted you to use a class instead of a struct, I did not request the use of private members nor did I ask for a friend relation, which is considerably worse as this class is only accessible in MissionAction.
Again, all I wanted is you to use a class with only public members, literally just commit the change I suggested.

Reply:
Actually, it's also accessible to vector. I've taken MissionDialog's one and only friend away. Now it is unloved and unwanted, like most other classes.

Reply:
that is not correct:

> A class definition should contain the following sections (most of which will be omitted for most classes) in this order, with each section introduced by a "public:", "protected:", or "private:" marker:
> 
>     public enums and nested classes
>     public static constants
>     public static functions
>     public functions
>     protected functions
>     private nested classes
>     private functions
>     private variables

Reply:
I clearly recall being quite confused in another PR when someone insisted I move the private class up to the top of the containing class's definition.

Reply:
well the style guide should be treated as the reference here

Reply:
I do myself sometimes forget to check it on my own PRs, thats what reviews are for

Reply:
I've moved the class down to its appropriate spot in the list.